event/consumer_test.go

```diff
@@ -2,138 +2,157 @@ package event_test
-				handlerWg.Wait()
+			<-eventHandler.EventUpdated
 
-				Convey("An event is sent to the mockEventHandler ", func() {
-					So(len(mockEventHandler.HandleCalls()), ShouldEqual, 1)
-					So(*mockEventHandler.HandleCalls()[0].HelloCalled, ShouldResemble, testEvent)
-				})
+			Convey("The expected event is sent to the handler", func() {
+				So(len(eventHandler.Events), ShouldEqual, 1)
 
-				Convey("The message is committed and the consumer is released", func() {
-					<-message.UpstreamDone()
-					So(len(message.CommitCalls()), ShouldEqual, 1)
-					So(len(message.ReleaseCalls()), ShouldEqual, 1)
-				})
+				actual := eventHandler.Events[0]
+				So(actual.DataType, ShouldEqual, expectedEvent.DataType)
+				So(actual.Title, ShouldEqual, expectedEvent.Title)
+			})
+			Convey("The message is committed and the consumer is released", func() {
+				<-message.UpstreamDone()
+				So(len(message.CommitCalls()), ShouldEqual, 1)
+				So(len(message.ReleaseCalls()), ShouldEqual, 1)
 			})
 		})
+		consumer.Close(nil)
+	})
+}
 
-		Convey("And two kafka messages, one with a valid schema and one with an invalid schema", func() {
+func TestConsumeWithFullBatchSizeMessage(t *testing.T) {
+	Convey("Given a consumer with a mocked message producer with an expected message", t, func() {
 
-			validMessage := kafkatest.NewMessage(marshal(testEvent), 1)
-			invalidMessage := kafkatest.NewMessage([]byte("invalid schema"), 0)
-			mockConsumer.Channels().Upstream <- invalidMessage
-			mockConsumer.Channels().Upstream <- validMessage
+		messageConsumer := kafkatest.NewMessageConsumer(false)
+		eventHandler := eventtest.NewEventHandler()
+		cfg, err := config.Get()
+		if err != nil {
+			t.Log(ctx, "failed to retrieve configuration", err)
+			t.Fail()
+		}
 
-			Convey("When consume messages is called", func() {
+		consumer := event.NewConsumer()
 
-				handlerWg.Add(1)
-				event.Consume(testCtx, mockConsumer, mockEventHandler, &config.Config{KafkaNumWorkers: 1})
-				handlerWg.Wait()
+		Convey("When consume is called", func() {
 
-				Convey("Only the valid event is sent to the mockEventHandler ", func() {
-					So(len(mockEventHandler.HandleCalls()), ShouldEqual, 1)
-					So(*mockEventHandler.HandleCalls()[0].HelloCalled, ShouldResemble, testEvent)
-				})
+			go consumer.Consume(testCtx, messageConsumer, eventHandler, cfg)
 
-				Convey("Only the valid message is committed, but the consumer is released for both messages", func() {
-					<-validMessage.UpstreamDone()
-					<-invalidMessage.UpstreamDone()
-					So(len(validMessage.CommitCalls()), ShouldEqual, 1)
-					So(len(invalidMessage.CommitCalls()), ShouldEqual, 1)
-					So(len(validMessage.ReleaseCalls()), ShouldEqual, 1)
-					So(len(invalidMessage.ReleaseCalls()), ShouldEqual, 1)
-				})
-			})
-		})
+			message1 := kafkatest.NewMessage([]byte(marshal(expectedEvent1)), 0)
+			messageConsumer.Channels().Upstream <- message1
+
+			message2 := kafkatest.NewMessage([]byte(marshal(expectedEvent)), 0)
```

Comment:
message2 to message 4 are the same messages - I don't think it makes sense instantiating different variables with the same content - unnecessary extra lines of code and adding to the time for running test as the expectedEvent is marshalled 4 times instead of just twice

Reply:
changed to 2 messages only. This was initially written to address max-size batch processing(having 4 events)  Now the default batch size is changed to 500 as recommended.